Avoid unnecessary QString heap allocations

Holzhaus opened the pull request.

Holzhaus committed:
Use QLatin1String for empty string literals

Fixes a bunch of -Wempty-qstringliteral warnings.
See https://github.com/KDE/clazy/blob/master/docs/checks/README-empty-qstringliteral.md for details.

Holzhaus committed:
rekordboxfeature: Avoid unneeded QString heap allocation

Fixes a -Wclazy-qstring-insensitive-allocation warning. See
https://github.com/KDE/clazy/blob/master/docs/checks/README-qstring-insensitive-allocation.md
for details.

Holzhaus committed:
wslidercomposed: Avoid temporary QString heap allocations

Fixes some of -Wclazy-qstring-ref warnings. See
https://github.com/KDE/clazy/blob/master/docs/checks/README-qstring-ref.md
for details.

Holzhaus committed:
GitHub Action: Add more clazy checks

daschuer commented on src/library/basesqltablemodel.cpp:
```diff
@@ -35,7 +35,7 @@ const int kMaxSortColumns = 3;
 const QString COLUMNS_SORTING = QStringLiteral("ColumnsSorting");
 
-const QString kEmptyString = QStringLiteral("");
+const QString kEmptyString = QLatin1String("");
```
This constant is useless and can be removed.

daschuer commented on src/preferences/dialog/dlgpreflv2.cpp:
```diff
@@ -61,7 +61,7 @@ DlgPrefLV2::DlgPrefLV2(QWidget* pParent, LV2Backend* lv2Backend,
         connect(button, SIGNAL(clicked()), this, SLOT(slotDisplayParameters()));
     }
-    effectNameLabel->setText(QStringLiteral(""));
+    effectNameLabel->setText(QLatin1String(""));
```
```suggestion
    effectNameLabel->clear());
```

daschuer commented on src/widget/wsearchlineedit.cpp:
```diff
@@ -19,7 +19,7 @@ const mixxx::Logger kLogger("WSearchLineEdit");
 const QColor kDefaultBackgroundColor = QColor(0, 0, 0);
 
-const QString kEmptySearch = QStringLiteral("");
+const QString kEmptySearch = QLatin1String("");
```
Can be removed completely

Holzhaus replied:
I was confused why this explicitly checks that the search string is not null, but there is not comment why this is necessary. I removed it and noticed not regression, but please give it a long, hard stare.

daschuer commented on src/library/dao/cuedao.cpp:
```diff
@@ -20,7 +20,7 @@ namespace {
 
 // The label column is not nullable!
-const QVariant kEmptyLabel = QVariant(QStringLiteral(""));
+const QVariant kEmptyLabel = QVariant(QLatin1String(""));
```
Please remove.

Holzhaus committed:
WSearchLineEdit: Remove unnecessary empty string constant

I didn't see comments indicating why an null QString is not possible in
this case. A manual test works fine, so we can probably remove this
unnecessary constant.

Holzhaus committed:
DlgPrefLv2: Use QLabel::clear() instead of setText(QLatin1String(""))

Holzhaus committed:
CueDAO: Inline constant that is only used in a single location

Holzhaus committed:
Merge branch '2.3' of github.com:mixxxdj/mixxx into clazy-qstring

Holzhaus committed:
Merge branch '2.3' of github.com:mixxxdj/mixxx into clazy-qstring

daschuer reviewed (approved):
Thank you. LGTM.

daschuer commented:
Waiting for CI

uklotzde commented:
Next one ;) LGTM, too.
